internal/retry/retry.go

```diff
@@ -184,8 +184,11 @@ func (r *Retryer) runRetryLoop(
 		// If this is the first time we've come across a failure to parse
 		// collection UUID, try again with the UUID elided (if the caller used
 		// RequestWithUUID).
-		if r.retryOnUUIDNotSupported && !r.aggregateDisallowsUUIDs && util.IsFailedToParseError(err) && util.HasServerErrorMessage(err, "collectionUUID") {
-			logger.Debug().Msg("Server version (< 5.0) does not support UUIDs in 'aggregate'. Will retry without UUID.")
+		if r.retryOnUUIDNotSupported && !r.aggregateDisallowsUUIDs &&
+			(util.IsFailedToParseError(err) && util.HasServerErrorMessage(err, "collectionUUID")) || // This error is returned in pre 5.0 server
```

Comment:
Can we split the logic into 2 variables, for example, `preV50collUUIDNotSupported` and `v50collUUIDNotSupported`, and add them to the debug log?

Reply:
Make sense. Done!